### PR TITLE
add script to test docs for warnings/errors

### PR DESCRIPTION
**Pull request by reidbarber**
This scripts loads every docs page in a browser and logs any runtime warnings/errors on page load.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 📝 Test Instructions:

`yarn test:docs`

Optionally specify the browser as an argument (i.e. `yarn test:docs webkit`), support `chromium` (default), `webkit`, and `firefox`.

### CODE REVIEW

**Comment by ktabors** [comment]
If you use something like puppeteer-cluster you should be able to parallelize. docs-differ crawls the site using this, but I didn't find much benefit to parallelization, probably because taking the screenshot was too intensive to gain much benefit from multiple threads.

**Comment by reidbarber** [comment]
@ktabors Yeah, was going to look into it to potentially speed it up a bit, but probably not needed since this doesn't need to be run too often.

**Review by snowystinger — DISMISSED** [review]
Nice, not nearly as many issues as I thought there'd be. Question, should the process finish when the errors are collected and done? It appears to stay active.

**Review by snowystinger — APPROVED** [review]

File: scripts/testDocs.js [hunk 1]
```diff
@@ -169,6 +169,7 @@ async function testDocs() {
     if (server && server.process) {
       server.process.kill();
     }
+    process.exit(0);
```
Comment (ktabors) [inline scripts/testDocs.js:172]: Thank you!!!

**Review by ktabors — APPROVED** [review]

File: scripts/testDocs.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      } else {
+        reject(new Error('Server did not start in time'));
+      }
+    };
+
+    checkServer();
+  });
+}
+
+async function getPageLinks() {
```
[the remark this answers is not in the document]
Comment (reidbarber) [reply scripts/testDocs.js:61]: Yeah, it seemed a little more organized and you can see the progress as it's running.

**Pull request merged**